docker/docker_transport_test.go

```diff
@@ -131,13 +169,25 @@ func TestReferenceDockerReference(t *testing.T) {
 		require.NotNil(t, dockerRef, c.input)
 		assert.Equal(t, c.dockerRef, dockerRef.String(), c.input)
 	}
+
+	for _, c := range ValidReferenceUnknownDigestTestCases {
```

Comment:
Isn’t this exactly the same code as for `validReferenceTestCases`? If so, there could just be one loop over `append(slices.Clone(validReferenceTestCases), validReferenceUnknownDigestTestCases…)`